### ISSUE TITLE
[DB] Project Setup Fails (macOS Mojave)

### ISSUE THREAD

**Issue by pimbuzan** [code4romania/seismic-risc#137]
Initializing the database fails on step: make init-env
(Added a log with the connection details)
Successfully built 2887f5a49024
Successfully tagged seismic-risc_client:latest
make init-db
docker-compose down -t 60
Removing network seismic-risc_default
WARNING: Network seismic-risc_default not found.
docker-compose run --rm api "./wait_for_db.py && ./manage.py migrate --no-input"
Creating network "seismic-risc_default" with the default driver
Creating seismic-risc_db_1 ... done
Connection details:
{'dbname': 'postgres', 'user': 'postgres', 'password': '', 'host': 'db', 'port': 5432}
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
DB not ready. Waiting for 1 second ...
Could not connect to DB within 30 seconds.
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/base/base.py", line 217, in ensure_connection
    self.connect()
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/base/base.py", line 195, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/postgresql/base.py", line 178, in get_new_connection
    connection = Database.connect(**conn_params)
  File "/usr/local/lib/python3.7/site-packages/psycopg2/__init__.py", line 126, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: could not translate host name "db" to address: Name or service not known

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "./manage.py", line 22, in <module>
    main()
  File "./manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/base.py", line 83, in wrapped
    res = handle_func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/commands/migrate.py", line 87, in handle
    executor = MigrationExecutor(connection, self.migration_progress_callback)
  File "/usr/local/lib/python3.7/site-packages/django/db/migrations/executor.py", line 18, in __init__
    self.loader = MigrationLoader(self.connection)
  File "/usr/local/lib/python3.7/site-packages/django/db/migrations/loader.py", line 49, in __init__
    self.build_graph()
  File "/usr/local/lib/python3.7/site-packages/django/db/migrations/loader.py", line 212, in build_graph
    self.applied_migrations = recorder.applied_migrations()
  File "/usr/local/lib/python3.7/site-packages/django/db/migrations/recorder.py", line 73, in applied_migrations
    if self.has_table():
  File "/usr/local/lib/python3.7/site-packages/django/db/migrations/recorder.py", line 56, in has_table
    return self.Migration._meta.db_table in self.connection.introspection.table_names(self.connection.cursor())
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/base/base.py", line 256, in cursor
    return self._cursor()
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/base/base.py", line 233, in _cursor
    self.ensure_connection()
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/base/base.py", line 217, in ensure_connection
    self.connect()
  File "/usr/local/lib/python3.7/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/base/base.py", line 217, in ensure_connection
    self.connect()
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/base/base.py", line 195, in connect
    self.connection = self.get_new_connection(conn_params)
  File "/usr/local/lib/python3.7/site-packages/django/db/backends/postgresql/base.py", line 178, in get_new_connection
    connection = Database.connect(**conn_params)
  File "/usr/local/lib/python3.7/site-packages/psycopg2/__init__.py", line 126, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
django.db.utils.OperationalError: could not translate host name "db" to address: Name or service not known

The env file generated:
(base) clj-mb-340:seismic-risc paul.imbuzan$ cat .env
# api
DJANGO_CONFIGURATION=Dev
DATABASE_URL=postgres://postgres@db:5432/postgres
# client
NODE_ENV=development

When running docker-compose up, there is another error message:

(base) clj-mb-340:seismic-risc paul.imbuzan$ docker-compose up
Starting seismic-risc_db_1 ... done
Creating seismic-risc-api  ... done
Creating seismic-risc-client ... done
Attaching to seismic-risc_db_1, seismic-risc-api, seismic-risc-client
seismic-risc-api | Connection details:
seismic-risc-api | {'dbname': 'postgres', 'user': 'postgres', 'password': '', 'host': 'db', 'port': 5432}
seismic-risc-client | + '[' development '!=' production ]
seismic-risc-client | + npm install
db_1      | Error: Database is uninitialized and superuser password is not specified.
db_1      |        You must specify POSTGRES_PASSWORD for the superuser. Use
db_1      |        "-e POSTGRES_PASSWORD=password" to set it in "docker run".
db_1      | 
db_1      |        You may also use POSTGRES_HOST_AUTH_METHOD=trust to allow all connections
db_1      |        without a password. This is *not* recommended. See PostgreSQL
db_1      |        documentation about "trust":
db_1      |        https://www.postgresql.org/docs/current/auth-trust.html
seismic-risc_db_1 exited with code 1

**Comment by tudoramariei** [comment]
@aramboi would you have the time to take a look over this?

**Comment by aramboi** [comment]
Tried to reproduce this on the same OS version, but could not. 

Make sure you have the latest Docker installed (currently stable 2.2.0.0) and that it has access to at least the default resources in the file sharing settings:

<img width="869" alt="Screenshot 2020-02-19 at 21 21 07" src="https://user-images.githubusercontent.com/12439/74872591-ccb99600-535d-11ea-8d19-f5f5577b27f3.png">

Something else you can try is `make redo-db` which should reinitialize the db.

**Comment by babasghenciu** [comment]
@tudoramariei @aramboi 
I had the same issue on Mac OS Catalina 10.15.3 (19D76). Latest version of Docker and anything else.

Basically if you setup up the project on a computer for the first time, you will have that problem. Apparently **the issue comes from latest Docker Images updates** ---> https://stackoverflow.com/questions/60262635/ci-cd-pipeline-with-postgresql-failed-with-database-is-uninitialized-and-superu

Adding ` environment:
      POSTGRES_HOST_AUTH_METHOD: trust` to my docker-compose file, 'fixed' the problem

**Comment by aramboi** [comment]
`POSTGRES_HOST_AUTH_METHOD=trust` is not preferable and it should work without it. 

@pimbuzan can you run the following command and let me know what it says:
```bash
docker-compose run db
```

The last message should be something similar to this:
```bash
2020-02-25 20:38:13.706 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2020-02-25 20:38:13.706 UTC [1] LOG:  listening on IPv6 address "::", port 5432
2020-02-25 20:38:13.710 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
2020-02-25 20:38:13.737 UTC [23] LOG:  database system was shut down at 2020-02-19 20:17:14 UTC
2020-02-25 20:38:13.765 UTC [1] LOG:  database system is ready to accept connections
```

**Comment by emanuelbesliu** [comment]
Managed to find a solution:

- `make drop-db`

- Add

`environment: - POSTGRES_HOST_AUTH_METHOD=trust`

not recommended but worked

- `make init-env`

- `docker-compose up`

**Comment by tudoramariei** [comment]
While #162 fixed this bug, it's not the proper fix, as per Angel's comment
Reopening and continuing to work on a fix
For having a chance to reproduce the bug, just remove the following lines from [docker-compose.yml](https://github.com/code4romania/seismic-risc/blob/master/docker-compose.yml)
```YAML
    environment:
      - POSTGRES_HOST_AUTH_METHOD=trust
```

**Comment by LucianU** [comment]
Why isn't `trust` an acceptable solution for development? The PosgreSQL container is not accessible from the outside, as far as I know.

**Comment by catileptic** [comment]
Closing this, it's fixed in #162
